### PR TITLE
add limits checks for power

### LINKED ISSUE #452: add limits tests for DC and AC power
This could be coarse like values > 0 + epsilon at night and values greater than 1.x times DC or AC capacity during the day. A separate function could be implemented for clear sky exceeded flags that's more careful (tracked in #200).

@cwhanse @wfvining I took a quick look at pvanalytics and didn't see any functions that addressed this. Did I miss something? Do you have any recommendations here?

For reference, #15 and #29 addressed limits checks for irradiance, temperature, and wind.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #452 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Playing around with how to set reasonable limits checks for AC and DC power. This code feels a bit arbitrary, but it also seems reasonable. Example below.

```python
latitude = 32.2
longitude = -110.9
elevation = 700

single = datamodel.SingleAxisModelingParameters(
    ac_capacity=1,
    dc_capacity=1.5,
    temperature_coefficient=-0.3,
    dc_loss_factor=0,
    ac_loss_factor=0,
    axis_tilt=0,
    axis_azimuth=0,
    ground_coverage_ratio=0.4,
    backtrack=True,
    max_rotation_angle=55
)

site = datamodel.SolarPowerPlant(
    name='Tucson, AZ',
    latitude=latitude,
    longitude=longitude,
    elevation=elevation,
    timezone='America/Phoenix',
    modeling_parameters=single
)

observation = datamodel.Observation(
    name='ac_power', variable='ac_power',
    interval_value_type='mean',
    interval_length=pd.Timedelta('15min'),
    interval_label='beginning', site=site, uncertainty=1)

start = pd.Timestamp('20200519 0700')
end = pd.Timestamp('20200520 0700')
index = pd.date_range(start=start, end=end, freq='15min', tz='UTC')

solar_position = pvmodel.calculate_solar_position(latitude, longitude, elevation, index)

cs = pvmodel.calculate_clearsky(latitude, longitude, elevation, solar_position['apparent_zenith'])

power = pvmodel.irradiance_to_power(single, solar_position['apparent_zenith'], solar_position['azimuth'], cs['ghi'], cs['dni'], cs['dhi'])

values = power.copy()
values.loc['20200519 0800':'20200519 0900'] = -0.1
values.loc['20200519 1000':'20200519 1100'] = 0.2
values.loc['20200519 1500':'20200519 1600'] = 1.5
values.loc['20200519 1900':'20200519 2000'] = -0.1
values.loc['20200519 2100':'20200519 2200'] = 1.1
values.loc['20200519 2300':] -= 0.02
validation_flags = tasks.validate_daily_ac_power(observation, values)
quality_flags = pd.Series(2, index=index)
for flag in validation_flags:
    quality_flags |= flag
data = pd.DataFrame({'value': values, 'quality_flag': quality_flags})
```

Screen Shot 2020-05-26 at 7 53 15 PM

### CODE REVIEW

**Comment by wholmgren** [comment]
@cwhanse What do you think of this approach? Useful? Too arbitrary?

**Comment by cwhanse** [comment]
> @cwhanse What do you think of this approach? Useful? Too arbitrary?

I think it will be useful to flag bogus data, or perhaps more often to warn when the capacity values are likely incorrect. It could have different limits for AC and DC power if it is worth distinguishing those quantities in this limit check. 30% over capacity for DC power is a somewhat generous upper bound, I'd set it to 20%. Anything more than a few (5%) percent over AC capacity should be suspect.

**Comment by wholmgren** [comment]
Thanks @cwhanse. The default limits for other variables are very generous, so I thought it would be better to err in that direction with the power limits. Maybe I went too far. Good point that upper limits on AC can be more strict than DC.

I'm wondering about API design for two different sets of limits. A couple of options:

* One private function that does the work, plus functions `check_ac_power_limits`, `check_dc_power_limits` that are the same except for default parameters.
* Keep a single function `check_power_limits`, and add a module-level dict `DEFAULT_POWER_LIMITS` for recommended defaults.

How would you prefer to use it (or eventually maintain in pvanalytics)? 

Any thoughts @alorenzo175?

**Comment by alorenzo175** [comment]
Seems like the first option with with separate functions is more consistent with the current functions, but if the single function is better for some other reason, that's fine.

Only other thing is that I would move the new code you've added to ``tasks.py`` into immediate validation functions for ac/dc power then call those in the daily validation like the ghi functions

**Comment by cwhanse** [comment]
> One private function that does the work, plus functions check_ac_power_limits, check_dc_power_limits that are the same except for default parameters.

I think this is more intuitive that having to identify the Series input as either DC or AC power.

File: solarforecastarbiter/validation/validator.py [hunk 1]
```diff
@@ -399,6 +399,115 @@ def check_rh_limits(rh, rh_limits=(0, 100)):
...
+    """ Checks for extreme AC power.
+
+    Parameters
+    ----------
+    power : Series
+        DC or AC power.
+    solar_zenith : Series
+        Solar zenith angle in degrees.
+    capacity : float
+        DC or AC capacity.
```
Comment (alorenzo175) [inline solarforecastarbiter/validation/validator.py:459]:
```suggestion
        DC capacity.
```

File: solarforecastarbiter/validation/validator.py [hunk 2]
```diff
@@ -399,6 +399,115 @@ def check_rh_limits(rh, rh_limits=(0, 100)):
...
```
Comment (alorenzo175) [inline solarforecastarbiter/validation/validator.py:416]:
```suggestion
        AC capacity.
```

**Pull request merged**